code/ProtectedT.h

```diff
@@ -36,8 +37,9 @@ struct protectedT
     {
         std::unique_lock<std::mutex> u_l(m);
         if (can_transform) {
-            auto f_t = mem_fn(f);
-            f_t(t, std::forward<Arg>(arg));
+            auto f_t = std::mem_fn(f);
+//            f_t(t, std::forward<Arg>(arg));
+            std::bind(f_t, t, std::forward<Arg>(arg));
```

Comment:
Here only a bind is done on the function. to work : 
auto f = std::bind(f_t, t, std::forward<Arg>(arg));
f();